src/plugins/formulas/autofill.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Registers on the formulas plugin instance.
+ *
+ * @param {object} pluginInstance The formulas plugin instance.
+ */
+export const registerAutofillHooks = (pluginInstance) => {
+  const lastAutofillSource = { value: undefined };
+
+  // Block autofill operation if at least one of the underlying's cell
+  // contents cannot be set, e.g. if there's a matrix underneath.
+  pluginInstance.addHook('beforeAutofill', (start, end) => {
```

Comment:
I'd leave the hook registering logic in the main file, and keep only the logic here. It would be consistent with the other plugins, and make it easier to assess what hooks is the plugin using at a quick glance.

Reply:
https://github.com/handsontable/handsontable/pull/7648/commits/90ebb0adca2e5679a31ae6388d1c8f980a813bb5